xdatasets/spatial.py

```diff
@@ -1,110 +1,128 @@
-from typing import Sequence, Tuple, Union, Dict, List, Optional
+import logging
+import warnings
 
-from clisops.core.subset import subset_shape, subset_time, create_mask, shape_bbox_indexer, subset_gridpoint
-from clisops.core.average import average_shape
+import pandas as pd
 import xarray as xr
+from clisops.core.subset import shape_bbox_indexer, subset_gridpoint
 from tqdm import tqdm
-import logging
-import xagg_no_xesmf_deps as xa
-import pandas as pd
 
 from .utils import HiddenPrints
 
+try:
+    import xagg as xa
+except ImportError:
+    warnings.warn("xagg is not installed. Please install it with `pip install xagg`")
+    xa = None
+
```

Comment:
@sebastienlanglois Are you OK with that and the change below that raises an exception if someone tries to use `create_weights_mask` without `xagg`?

Reply:
Yes I'm good with that